src/validation.cpp

```diff
@@ -5568,17 +5568,23 @@ double GuessVerificationProgress(const ChainTxData& data, const CBlockIndex *pin
         return 0.0;
     }
 
-    int64_t nNow = time(nullptr);
+    int64_t end_of_chain_timestamp = pindex->GetBlockTime();
```

Comment:
Hi, I'm not sure if I understood you correctly, but I guess you say it because it's possible to call the function with ``hindex = nullptr``.

To solve that I propose this changes:

```c++
double GuessVerificationProgress(const ChainTxData& data, const CBlockIndex* pindex, const CBlockIndex* hindex) {
    if (pindex == nullptr || hindex == nullptr)
        return 0.0;
...
    int64_t end_of_chain_timestamp;

    if (hindex->nChainWork > pindex->nChainWork) {
        int64_t header_age = time(nullptr) - hindex->GetBlockTime();
        end_of_chain_timestamp = header_age < 24 * 60 * 60 ? std::max(pindex->GetBlockTime(), hindex->GetBlockTime()) : pindex->GetBlockTime();
    } else {
        end_of_chain_timestamp = pindex -> GetBlockTime();
    }
...
```

Please let me know your thoughts

Reply:
Oh, understood, yes, I think so.
But it's the way to not move the function inside ``chainman`` and keep the actual structure. Do you think is more reasonable move it inside?